<?xml version="1.0"?>
<launch>

<arg name="adaptVideoParamsEnabled" default="false"/>

<group ns="/Vehicle/Video">

<node name="RtspServer" type="VehicleRtspServer" pkg="tod_video" output="screen">
    <param name="bitrate" value="1000"/>
</node>

<node name="ConfigsReceive" type="VehicleConfigsReceive" pkg="tod_video" output="screen"
    if="$(eval adaptVideoParamsEnabled)">
    <param name="debug" value="false"/>
</node>

<node pkg="tod_video" name="BandwidthManager" type="VehicleBandwidthManager" output="screen"
    if="$(eval adaptVideoParamsEnabled)">
    <param name="debug" value="false"/>
</node>

<node name="PredQoSClient" type="VehiclePredQoSClient" pkg="tod_video" output="screen" if="$(eval adaptVideoParamsEnabled)">
    <rosparam command="load" file="$(find tod_video)/yaml/bandwidth_map.yaml"/>
    <param name="debug" value="false"/>
</node>

<node name="BitratePredictionSender" type="VehicleBitratePredictionSender" pkg="tod_video" output="screen" if="$(eval adaptVideoParamsEnabled)"/>

</group> <!--group: /Vehicle/Video-->

</launch>
